ntoskrnl/ps/psmgr.c

```diff
@@ -212,8 +212,8 @@ PsLocateSystemDll(VOID)
     }
 
     /* Check if the image is valid */
-    Status = MmCheckSystemImage(FileHandle, TRUE);
-    if (Status == STATUS_IMAGE_CHECKSUM_MISMATCH)
+    Status = MmCheckSystemImage(FileHandle);
+    if (Status == STATUS_IMAGE_CHECKSUM_MISMATCH || Status == STATUS_INVALID_IMAGE_PROTECT)
```

Comment:
Can't `STATUS_IMAGE_MP_UP_MISMATCH` happen here too?
Do we actually care for the error value? Shouldn't we check for (not) success? (everywhere)

Reply:
This doesn't matter for user-mode modules (such as ntdll.dll) because they don't directly use _kernel_ locking mechanisms that would change between UP and MP builds.

Reply:
Alright for this specific status/case, but why are other (propagated `Status`) errors ignored? (everywhere)

Reply:
Yes, here cares for those two error codes only, other places I didn't go through yet.

Reply:
IMO, here the caller determine which error code are fatal is make sense.